src/Routing/Route/DashedRoute.php

```diff
@@ -56,11 +56,12 @@ protected function _camelizePlugin($plugin)
      * camelBacked form.
      *
      * @param string $url The URL to parse
+     * @param string $method The HTTP method.
      * @return array|false An array of request parameters, or false on failure.
      */
-    public function parse($url)
+    public function parse($url, $method = '')
```

Comment:
While the new argument has a default param wouldn't it still throw a notice in an existing class which has overridden this method?